packages/explat-client/src/internal/test/requests.ts

```diff
@@ -0,0 +1,233 @@
+/**
+ * External dependencies
+ */
+// This is required to fix the "regeneratorRuntime is not defined" error
+import '@automattic/calypso-polyfills';
+
+/**
+ * Internal dependencies
+ */
+import * as Timing from '../timing';
+import * as Requests from '../requests';
+import type { Config, ExperimentAssignment } from '../../types';
+import { delayedValue, validExperimentAssignment } from '../test-common';
+import * as ExperimentAssignments from '../experiment-assignments';
+
+const spiedMonotonicNow = jest.spyOn( Timing, 'monotonicNow' );
+
+const mockedFetchExperimentAssignment = jest.fn();
+const mockedGetAnonId = jest.fn();
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore; Not using the full Config
+const mockedConfig: Config = {
+	fetchExperimentAssignment: mockedFetchExperimentAssignment,
+	getAnonId: mockedGetAnonId,
+};
```

Comment:
Since mocking / overriding the other two items is trivial, I think we might as well do it rather than disable the warnings. Something like:

```suggestion
const mockedConfig: Config = {
	fetchExperimentAssignment: mockedFetchExperimentAssignment,
	getAnonId: mockedGetAnonId,
	logError: () => {},
	isDevelopmentMode: false,
};
```

With this, we're also more explicit about not testing development mode.